skimage/morphology/convex_hull.py

```diff
@@ -106,6 +106,13 @@ def convex_hull_image(image, offset_coordinates=True, tolerance=1e-10):
     .. [1] https://blogs.mathworks.com/steve/2011/10/04/binary-image-convex-hull-algorithm-notes/
 
     """
+    image_shape = np.array(image.shape)
+    # Look for 1D axes
+    axes_1D = np.where(image_shape == 1)[0]
+    label_is_2D = (axes_1D.size != 0) and (len(image_shape) == 3)
+    if label_is_2D:
+        # If there is at least one axis 1D and image is 3D, treat it as 2D
+        image = np.squeeze(image, axis=axes_1D[0])
```

Comment:
Hi @stefanv, hi @lagru,

Thanks for the comments, I will address them and get back to you here soon.

Reply:
> Why not just squeeze every dimension of size 1 here? E.g. that way it would work on an image of shape (1, 3, 1, 4) too and the code is a lot more straight forward.
> 
> Also I think a comment on "why do we reduce this" would be useful down the line as well.
> 
> How about
> 
> and just call `mask = mask.reshape(original_image_shape)` below?

If I remember correctly, that was my initial solution as well, but this would create problems if the label is 1D, for example, a straight line label in a 3D image. I guess that's also why I added the `SAMPLE_3D_LABEL_1D` test.

Unfortunately, I am currently unable to test this again because of installation issues #6494

Reply:
Hmm, I'm not sure I follow. E.g. this
```python
image = np.zeros((9, 5, 1), dtype=bool)
image[2:7, 3, 0] = True
mask = ski.morphology.convex_hull_image(image)
np.nonzero(mask)  # returns (array([2, 3, 4, 5, 6]), array([3, 3, 3, 3, 3]))
```
works as I would expect (it doesn't with the current version on `main`!). Though there is an issue with my suggestion for the special case of an image with only one pixel. By chance this worked for images with the shape `(1, 1)` because then it would take the route via https://github.com/scikit-image/scikit-image/blob/8803f65854517c06247b654e2c8b31d1d28e202d/skimage/morphology/convex_hull.py#L113-L114 which would return `coords` such that `ConvexHull` is able to deal with it.  However, we could address this special case by adding another condition for `image.size == 1`. E.g.
```python
    if ndim == 2:
        coords = possible_hull(np.ascontiguousarray(image, dtype=np.uint8))
    elif image.size == 1:
        # return early for trivial case that isn't supported by ConvexHull
        return np.ones(original_image_shape, dtype=bool)
    else:
        coords = np.transpose(np.nonzero(image))
```

Reply:
Sadly I don't think it's as easy as returning `np.ones(original_image_shape, dtype=bool)`. E.g. an image such as `array([[False,  True, False,  True, False]])` (`shape == (1, 5)`) should work as well.

 I think you got the idea the first time: squeeze every axis such that image is still at least 2D. The logic / processing flow in this function is a bit of a mess and the proposed fix seems a bit hacky. But still, I think our best bet is to fix it this way, and add a generous amount of tests and comments.